File: js/8kyu/enumerable_magic_#1_true_for_all.js
// Kata url: https://www.codewars.com/kata/54598d1fcbae2ae05200112c

function all(arr, fun) {
    for (let i = 0; i < arr.length; i++) {
        if (!fun(arr[i])) return false;
    }
    return true;
}

File: python/8kyu/Enumerable_magic_#1_true_for_all.py
# Kata url: https://www.codewars.com/kata/54598d1fcbae2ae05200112c.

from typing import Callable, Any, List


def all(seq: List[Any], fun: Callable[[Any], bool]) -> bool:
    return sum(fun(x) for x in seq) == len(seq)